### PR TITLE
[sc-17485] Update catalog for Azure AD

### PR DESCRIPTION
**Pull request by Mohitp98**
Updated Azure AD catalog with new entities and the relationship between them:
- `Role`
- `RoleMember`

### CODE REVIEW

**Review by r-rakshith — CHANGES_REQUESTED** [review]

File: azure-ad.sgnl.yaml [hunk 1]
```diff
@@ -573,6 +573,54 @@ entities:
...
+        type: String
+      - name: roleTemplateId
+        externalId: roleTemplateId
+        type: String
+      - name: deletedDateTime
+        externalId: deletedDateTime
+        type: DateTime
+  RoleMember:
+    # https://learn.microsoft.com/en-us/graph/api/directoryrole-list-members?view=graph-rest-1.0    
+    description: "Directory Role Member Entity in Azure AD"
```
Comment (arooxa) [inline azure-ad.sgnl.yaml:604]: nit: This is the only catalog that has quotes around the description, we should remove this for all entities

File: azure-ad.sgnl.yaml [hunk 2]
```diff
@@ -586,6 +634,16 @@ relationships:
     displayName: User Member
     fromAttribute: GroupMember.memberId
     toAttribute: User.id
+  RoleMember:
```
Comment (r-rakshith) [inline azure-ad.sgnl.yaml:637]: Please reorganize this and put this above `UserMember`. Let's keep similar things grouped together and not intertwine relationships between different entities for readability.

And as requested in my previous comments, please add the role equivalents to the following relationships
```yaml
UserMember:
    name: Member
    displayName: User Member
    fromAttribute: GroupMember.memberId
    toAttribute: User.id
  GroupMember:
    name: Member
    displayName: Group Member
    fromAttribute: GroupMember.memberId
    toAttribute: Group.id
  MemberOf:
    name: MemberOf
    displayName: Member Of
    fromAttribute: GroupMember.groupId
    toAttribute: Group.id
  UserMemberGroup:
    name: Group
    displayName: User Member Group
    path:
      - relationship: UserMember
        direction: Backward
      - relationship: MemberOf
        direction: Forward
  GroupMemberGroup:
    name: Group
    displayName: Group Member Group
    path:
      - relationship: GroupMember
        direction: Backward
      - relationship: MemberOf
        direction: Forward
```

**Review by r-rakshith — CHANGES_REQUESTED** [review]

File: azure-ad.sgnl.yaml (hunk 2 above)
Comment (Mohitp98) [reply azure-ad.sgnl.yaml:628]: Please check now

File: azure-ad.sgnl.yaml [hunk 3]
```diff
@@ -581,6 +625,24 @@ relationships:
     toAttribute: User.id
     # [TODO sc-11020] Give these a unique ID, rename both of these to Member,
     # and keep corresponding path relationships unique below
+  RoleMember:
+    name: MemberOf
+    displayName: Role To Role Member
+    fromAttribute: Role.id
+    toAttribute: RoleMember.roleId
```
Comment (r-rakshith) [inline azure-ad.sgnl.yaml:632]: Please make sure to keep the name, displayName, fromAttribute and toAttribute consistent with the GroupMember counterparts.

For example, 
a `GroupMember` relationship equivalent for a role would look like
```suggestion
  RoleMember:
    name: Member
    displayName: RoleMember
    fromAttribute: RoleMember.memberId
    toAttribute: Role.roleId
```


Please make similar changes to ALL relationships listed in https://github.com/SGNL-ai/catalog-sgnl-templates/pull/49/files#r1541841918

**Comment by r-rakshith** [comment]
Please let me know once you've tested the catalog on DEV. I'll review and approve after that.

**Comment by Mohitp98** [comment]
> Please let me know once you've tested the catalog on DEV. I'll review and approve after that.

Yes, It is tested, attaching relationship details below:

image
image

**Review by r-rakshith — APPROVED** [review]
LGTM

**Pull request merged**